File: 551. Student Attendance Record I/Java/Solution.java
public class Solution {
    public boolean checkRecord(String s) {
        boolean containsA = false;
        for (int i = 0; i < s.length(); i++) {
            char letter = s.charAt(i);
            if ((letter == 'A' && containsA) || (letter == 'L' && i >= 2 && s.charAt(i - 1) == 'L' && s.charAt(i - 2) == 'L')) {
                return false;
            }
            if (letter == 'A' && !containsA) {
                containsA = true;
            }
        }
        return true;
    }
}

File: 551. Student Attendance Record I/Python/Solution.py
class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        contains_A = False
        for i, letter in enumerate(s):
            if (letter == 'A' and contains_A) or (letter == 'L' and i >= 2 and s[i - 1] == 'L' and s[i - 2] == 'L'):
                return False
            if letter == 'A' and not contains_A:
                contains_A = True
        return True
